test/e2e/proxy_test.go

```diff
@@ -77,35 +80,113 @@ func (tc *testContext) testCerts(t *testing.T) {
-					// Get the number of existing certs equivalent to the expected cert
-					"(Get-ChildItem -Path Cert:\\LocalMachine\\Root | Where-Object {$expectedCert.Equals($_)}).Count",
-					expectedCertBase64)
-				out, err := tc.runPowerShellSSHJob(fmt.Sprintf("get-cert-%d", i), commandToRun, addr)
-				if err != nil {
-					require.NoError(t, err, "error running SSH job: %w", err)
-				}
-				// Final line should contain a single number representing the number of certs found equal to the target
-				count, err := strconv.Atoi(finalLine(out))
-				require.NoError(t, err)
+			proxyCertsImported, err := tc.checkProxyCerts(addr, []byte(trustedCABundle), 1)
+			require.NoError(t, err, "error determining if proxy certs are imported")
+			assert.True(t, proxyCertsImported, "proxy certs not imported")
+		})
+	}
+}
+
+// testCertsRemoval tests that any additional certificates from the proxy's trusted bundle are removed by each node
+func (tc *testContext) testCertsRemoval(t *testing.T) {
+	// TODO: this only tests the user-provided certs, a subset of the required proxy certificates.
+	// Should be addressed with https://issues.redhat.com/browse/WINC-1144
+	cm, err := tc.client.K8s.CoreV1().ConfigMaps(userCABundleNamespace).Get(context.TODO(), userCABundleName, meta.GetOptions{})
+	require.NoErrorf(t, err, "error getting user-provided CA ConfigMap: %w", err)
+
+	// Read all expected certs from CM data
+	trustedCABundle := cm.Data[certificates.CABundleKey]
+	assert.Greater(t, len(trustedCABundle), 0, "no additional user-provided certs in bundle")
+
+	require.NoError(t, tc.removeProxyTrustedCA(), "error removing user CA bundle from cluster-wide proxy")
+	require.NoError(t, tc.waitForAllNodesToReboot())
+
+	for _, node := range gc.allNodes() {
+		t.Run(node.GetName(), func(t *testing.T) {
+			addr, err := controllers.GetAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
+
+			proxyCertsRemoved, err := tc.checkProxyCerts(addr, []byte(trustedCABundle), 0)
+			require.NoError(t, err, "error determining if proxy certs are removed")
+			assert.True(t, proxyCertsRemoved, "proxy certs not removed")
+		})
+	}
+}
 
-				assert.Equalf(t, count, 1, "unexpected cert %d count on node %s: expected 1, found %d", i, node, count)
-				i++
+// waitForAllNodesToReboot waits for all Windows nodes in the cluster to be rebooted
+func (tc *testContext) waitForAllNodesToReboot() error {
+	// Wait for for reboot annotation to be applied to all Windows nodes, and then wait for removal
+	if err := tc.waitForRebootAnno(false); err != nil {
+		return err
+	}
+	return tc.waitForRebootAnno(true)
+}
+
+// waitForRebootAnno waits for all nodes in the cluster to have the reboot annotation either applied or removed
+func (tc *testContext) waitForRebootAnno(waitingForRemoval bool) error {
+	allWinNodes := make(map[string]struct{})
+	for _, node := range gc.allNodes() {
+		allWinNodes[node.Name] = struct{}{}
+	}
```

Comment:
Wouldn't it be safer to create this map before removing the certs from the cluster-wide proxy configuration?

Reply:
I don't see why as gc.allNodes will remain as is (even if WMCO were to implode and delete all node objects)

I think the only way the gc's nodes get updated is by calling `loadExistingNodes()` or `loadwaitForConfiguredWindowsNodes()`

Reply:
OK, then it leads to another question... don't we need to update gc.Nodes between `waitForRebootAnnotation()` calls?